common_factories.py

```diff
@@ -543,3 +556,44 @@ def getRpmAutobakeFactory(mtrDbPool):
         )
     )
     return f_rpm_autobake
+
+
+def getAsanUbsanFactory(mtrDbPool, ubsan=False):
+    ## f_rpm_autobake
+    f_asan_ubsan_build = getBuildFactoryPreCompile()
+
+    f_asan_ubsan_build.addStep(
+        steps.ShellCommand(
+            command='echo "leak:libtasn1\nleak:libgnutls\nleak:libgmp" > mysql-test/lsan.supp',
+            doStepIf=filterBranch,
+        )
+    )
+    f_asan_ubsan_build.addStep(
+        steps.ShellCommand(command="cat mysql-test/lsan.supp", doStepIf=filterBranch)
+    )
+
+    additional_args = "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DWITH_ASAN=YES -DPLUGIN_ROCKSDB=NO -DPLUGIN_CONNECT=NO -DWITH_SAFEMALLOC=OFF -DWITH_ZLIB=bundled -DWITH_SSL=bundled -DWITH_DBUG_TRACE=OFF -DWITH_SAFEMALLOC=OFF"
```

Comment:
can we have options one above the other (reduce the line), it would help maintenance.

Reply:
maybe with a list though it's not ideal

Reply:
```console
additional_args = "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
  -DWITH_ASAN=YES \
  ...
```

Something like that won't work,

Reply:
I don't like that since you always need to be careful to leave that extra space at the end between the last character and \. So, I would say a list is safer

Reply:
up to you